iceoryx_binding_c/source/cpp2c_subscriber.cpp

```diff
@@ -13,18 +13,57 @@
 // limitations under the License.
 
 #include "iceoryx_binding_c/internal/cpp2c_subscriber.hpp"
+#include "iceoryx_binding_c/enums.h"
+#include "iceoryx_binding_c/internal/cpp2c_enum_translation.hpp"
+#include "iceoryx_posh/internal/popo/ports/subscriber_port_user.hpp"
 
-void cpp2c_Subscriber::setConditionVariable(iox::popo::ConditionVariableData* const conditionVariableDataPtr) noexcept
+cpp2c_Subscriber::~cpp2c_Subscriber()
 {
-    iox::popo::SubscriberPortUser(m_portData).setConditionVariable(conditionVariableDataPtr);
+    if (m_portData)
+    {
+        iox::popo::SubscriberPortUser(m_portData).destroy();
+    }
 }
 
-bool cpp2c_Subscriber::hasTriggered() const noexcept
+iox_WaitSetResult
+cpp2c_Subscriber::attachToWaitset(iox::popo::WaitSet& waitset,
+                                  const iox_SubscriberEvent subscriberEvent,
+                                  const uint64_t triggerId,
+                                  const iox::popo::Trigger::Callback<cpp2c_Subscriber> callback) noexcept
 {
-    return iox::popo::SubscriberPortUser(m_portData).hasNewChunks();
+    static_cast<void>(subscriberEvent);
```

Comment:
So we would differentiate here if there will be additional subscriberEvents?

Reply:
Exactly.